bazel/grpc_build_system.bzl

```diff
@@ -298,39 +310,58 @@ def grpc_cc_test(name, srcs = [], deps = [], external_deps = [], args = [], data
         "linkstatic": linkstatic,
     }
     if uses_polling:
-        # the vanilla version of the test should run on platforms that only
-        # support a single poller
-        native.cc_test(
-            name = name,
-            testonly = True,
-            tags = (tags + [
-                "no_linux",  # linux supports multiple pollers
-            ]),
-            **args
-        )
-
-        # on linux we run the same test multiple times, once for each poller
-        for poller in POLLERS:
-            native.sh_test(
-                name = name + "@poller=" + poller,
-                data = [name] + data,
-                srcs = [
-                    "//test/core/util:run_with_poller_sh",
-                ],
-                size = size,
-                timeout = timeout,
-                args = [
-                    poller,
-                    "$(location %s)" % name,
-                ] + args["args"],
-                tags = (tags + ["no_windows", "no_mac"]),
-                exec_compatible_with = exec_compatible_with,
-                exec_properties = exec_properties,
-                shard_count = shard_count,
-                flaky = flaky,
-            )
+        # When true, EventEngines will only be exercised against one poller
+        engine_only = False
+        for engine in EVENT_ENGINES:
+            if "no_windows" not in engine["tags"] and "no_mac" not in engine["tags"]:
+                # the vanilla version of the test should run on platforms that only
+                # support a single poller
+                native.cc_test(
+                    name = name,
+                    testonly = True,
+                    tags = (tags + engine["tags"] + [
+                        "no_linux",  # linux supports multiple pollers
```

Comment:
Can you help me understand this comment? What does it mean to add "no_linux" here and how is this comment explaining that?

Reply:
It's an artifact of the legacy I/O architecture, iomgr. There are multiple iomgr implementations, and for any tests that rely on I/O, this logic exercises every known poller. On non-linux platforms, there happens to be one iomgr implementation per platform. For linux, there are three in OSS: poll, epoll1, and epollex. This logic says "for non-linux platforms, we can run this test thusly" without any of the fiddly configuration done below for linux, and our CI uses these tags to determine which tests to run on which platforms. Please note that this comment is not new, and that this logic will be simplified when we delete iomgr.